### ISSUE TITLE
Building for iOS

### ISSUE THREAD

**Issue by snap9** [openssl/openssl#2194]
What is exactly are the `ios-cross`, `ios64-cross`, and `iphoneos-cross` platforms?

Which of them should I use to build OpenSSL for iOS?

Some people claim that I also need to build with `darwin64-x86_64-cc` in order for my iOS app to work in the Simulator. Is this true? (I am only on 64-bit Macs, so the the other "darwins" seem irrelevant.)

**Comment by dot-asm** [comment]
Fist of all you have to recognize that there are contributed platforms, i.e. ones that we don't have "iron grip" over, which in turn means that we're likely to fail to answer all questions. I can tell what's ios-cross is (target for 32-bit ARM-based iOS devices) and ios64-cross (target for 64-bit ARM-based iOS devices), but not much about iphoneos-cross (ought to be target for 32-bit ARM-based older iPhone devices, but it's not in master branch), or what it takes to target "The Simulator". darwin target might and probably has good chance to work, but I don't make any promises.

I'm leaving this open, but if nobody (presumably from public, as to my knowledge no team member works with "The Simulator") chimes in and provides some wisdom, it would have to be closed...

**Comment by snap9** [comment]
@dot-asm Thanks for the useful info. About `iphoneos-cross`, what do you mean it's not in master? I downloaded openssl-1.1.0c and in `./Configure LIST`, `iphoneos-cross` is in that list. Hasn't the meaning of these platforms been documented anywhere?

**Comment by dot-asm** [comment]
> About iphoneos-cross, what do you mean it's not in master?

Oh! I thought that it was removed, but apparently not, sorry about confusion. It's legacy config line (should be obvious because originally it was called iPhoneOS), which means that you should be discouraged to use it.

**Comment by oneton** [comment]
There is a script that can build OpenSSL for iOS: https://github.com/x2on/OpenSSL-for-iPhone

I adapted the 1.0.2 version to the new OpenSSL 1.1.0 build system. The iOS simulator indeed requires a Darwin x64 build. I created a separate configuration file (based on the default OpenSSL build targets for Darwin and iOS) for iOS/tvOS devices and simulators. https://github.com/x2on/OpenSSL-for-iPhone/blob/master/config/20-ios-tvos-cross.conf

The most important additional option for the x64 simulator is a reference to build using the iOS SDK.

However, a few people experience problems when using the script's build for armv7 (e.g. iPhone 5) https://github.com/x2on/OpenSSL-for-iPhone/issues/131
